packages/world/ts/actions/delegation.ts

```diff
@@ -0,0 +1,56 @@
+import type { WalletClient, Transport, Chain, Account, Hex, WalletActions, WriteContractReturnType } from "viem";
+import { getAction, getAbiItem, formatAbiItem, toFunctionSelector, type GetAbiItemParameters } from "viem/utils";
+import { writeContract } from "viem/actions";
+import { encodeSystemCallFrom, type SystemCallFrom } from "../encodeSystemCallFrom";
+
+type DelegationParameters = {
+  worldAddress: Hex;
+  delegatorAddress: Hex;
+  getSystemId: (functionSelector: Hex) => Hex;
+};
+
+// By extending clients with this function after delegation, the delegation automatically applies
+// to the World contract writes, meaning these calls are made on behalf of the delegator.
+export function delegation<TChain extends Chain, TAccount extends Account>({
+  worldAddress,
+  delegatorAddress,
+  getSystemId,
+}: DelegationParameters): (
+  client: WalletClient<Transport, TChain, TAccount>,
+) => Pick<WalletActions<TChain, TAccount>, "writeContract"> {
+  return (client) => ({
+    // Applies to: `client.writeContract`, `getContract(client, ...).write`
+    writeContract: (originalArgs): Promise<WriteContractReturnType> => {
```

Comment:
```suggestion
    writeContract: (writeArgs): Promise<WriteContractReturnType> => {
```
`writeArgs` or `writeContractArgs` might be a little more clear when we're juggling with `callFromArgs`

Reply:
Right, `writeArgs` has more info in the name, thus it is better.